controllers/module_reconciler.go

```diff
@@ -305,40 +285,14 @@ func (r *ModuleReconciler) handleBuild(ctx context.Context,
 	return buildRes.Requeue, nil
 }
 
-// is sign defined in the CR
-func (r *ModuleReconciler) willSign(mod *kmmv1beta1.Module, km *kmmv1beta1.KernelMapping) bool {
-	if mod.Spec.ModuleLoader.Container.Sign == nil && km.Sign == nil {
-		return false
-	}
-	return true
-}
-
-// is build defined in the CR
-func (r *ModuleReconciler) willBuild(mod *kmmv1beta1.Module, km *kmmv1beta1.KernelMapping) bool {
-	if mod.Spec.ModuleLoader.Container.Build == nil && km.Build == nil {
-		return false
-	}
-	return true
-}
-
-// append tag to the image name cleanly, avoiding messing up the name or getting "name:-tag"
-func (r *ModuleReconciler) appendToTag(name string, tag string) string {
-	if strings.Contains(name, ":") {
-		return name + "_" + tag
-	} else {
-		return name + ":" + tag
-	}
-
-}
-
 func (r *ModuleReconciler) handleSigning(ctx context.Context,
 	mod *kmmv1beta1.Module,
 	km *kmmv1beta1.KernelMapping,
 	kernelVersion string,
 	containerImage string) (bool, error) {
 
 	// if sign is not specified skip
-	if !r.willSign(mod, km) {
+	if !module.ShouldBeSigned(mod, km) {
```

Comment:
Why checkImageExists removed from handleBuild and left in handleSigning?

Reply:
There are 2 reasons why this is for the time being not refactored yet:

- I wanted to confirm the refactoring effort with the `handleBuild` business logic first and then move to the `handleSignging` refactoring
- there is an open [question](https://github.com/kubernetes-sigs/kernel-module-management/pull/156#discussion_r1026542716) about where to host the `checkIfImageExists` method/function

So I will soon come back to this one. Does this make sense?

Reply:
Sure! Are you going to do it in a different PR, or later in this one?

Reply:
I'm targeting this PR. As soon as we have covered the rest of the review comments, I'll go for it.